Mage.Sets/src/mage/cards/t/TheMillenniumCalendar.java

```diff
@@ -0,0 +1,150 @@
+        this.addAbility(new TheMilleniumCalendarTriggeredAbility());
+
+        // {2}, {T}: Double the number of time counters on The Millennium Calendar.
+        Ability ability = new SimpleActivatedAbility(
+                new DoubleCountersSourceEffect(CounterType.TIME),
+                new ManaCostsImpl<>("{2}")
+        );
+        ability.addCost(new TapSourceCost());
+        this.addAbility(ability);
+
+        // When there are 1,000 or more time counters on The Millennium Calendar, sacrifice it and each opponent loses 1,000 life.
+        this.addAbility(new TheMillenniumCalendarStateTriggeredAbility());
+    }
+
+    private TheMillenniumCalendar(final TheMillenniumCalendar card) {
+        super(card);
+    }
+
+    @Override
+    public TheMillenniumCalendar copy() {
+        return new TheMillenniumCalendar(this);
+    }
+}
+
+class TheMilleniumCalendarTriggeredAbility extends TriggeredAbilityImpl {
+
+    public TheMilleniumCalendarTriggeredAbility() {
+        super(Zone.BATTLEFIELD, null, false);
+    }
+
+    protected TheMilleniumCalendarTriggeredAbility(final TheMilleniumCalendarTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public TheMilleniumCalendarTriggeredAbility copy() {
+        return new TheMilleniumCalendarTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.UNTAPPED_BATCH;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (!game.isActivePlayer(getControllerId())) {
+            return false;
+        }
+        UntappedBatchEvent batchEvent = (UntappedBatchEvent) event;
+        int count = batchEvent
+                .getEvents()
+                .stream()
+                .filter(UntappedEvent::isAnUntapStepEvent)
+                .map(UntappedEvent::getTargetId)
+                .map(game::getPermanent)
+                .filter(Objects::nonNull)
+                .filter(p -> p.getControllerId().equals(getControllerId()))
+                .mapToInt(p -> 1)
+                .sum();
+
+        if (count <= 0) {
+            return false;
+        }
+
+        this.getEffects().clear();
+        this.addEffect(new AddCountersSourceEffect(CounterType.TIME.createInstance(count)));
+        this.getEffects().setValue(infoKey, count); // for the trigger on the stack to display the info.
+        return true;
+    }
+
+    private static final String infoKey = "number_untapped";
+
+    @Override
+    public String getRule() {
+        // this trigger might want to expose extra info on the stack
+        String triggeredInfo = "";
+        if (!this.getEffects().isEmpty()) {
+            triggeredInfo += "<br><br><i>Number of untapped permanents: " + this.getEffects().get(0).getValue(infoKey) + "</i>";
```

Comment:
I've been using that way to provide info to the displayed trigger a few times now.
This is more for opening a discussion, but we could use a more explicit way to achieve that? Similar to a hint, but only for the trigger in the trigger order panel / on the stack.

Reply:
Card hints can be used to stack objects too, search `Lose life amount` for example:
![shot_231029_042117](https://github.com/magefree/mage/assets/8344157/7fae00ac-802e-4fb0-9f1c-8f6b992bd715)

Reply:
That makes a lot of sense! I'll use that instead of the getValue/setValue. Way less adhoc.